TESTING.md

```diff
@@ -0,0 +1,53 @@
+<!-- TOC -->
+- [Testing](#testing)
```

Comment:
Can you add links to all the subsections of the tree? 
It would be helpful to navigate.

Refer to UPGRADING.md if you need an example.

Reply:
Thank you, good catch! I will add all the subsections.

Reply:
Added all the headers so far.